packages/react-native-editor/src/setup.js

```diff
@@ -0,0 +1,140 @@
+	validateThemeColors,
+	validateThemeGradients,
+} from '@wordpress/block-editor';
+import { unregisterBlockType, getBlockType } from '@wordpress/blocks';
+import { addAction, addFilter } from '@wordpress/hooks';
+import * as wpData from '@wordpress/data';
+import { initializeEditor } from '@wordpress/edit-post';
+import { registerCoreBlocks } from '@wordpress/block-library';
+
+let editorComponent;
+let blocksRegistered = false;
+
+const reactNativeSetup = () => {
+	// Disable warnings as they disrupt the user experience in dev mode
+	// eslint-disable-next-line no-console
+	console.disableYellowBox = true;
+
+	I18nManager.forceRTL( false ); // Change to `true` to debug RTL layout easily.
+};
+
+const gutenbergSetup = () => {
+	// wp-data
+	const userId = 1;
+	const storageKey = 'WP_DATA_USER_' + userId;
+	wpData.use( wpData.plugins.persistence, { storageKey } );
+
+	setupApiFetch();
+
+	const isHermes = () => global.HermesInternal !== null;
+	// eslint-disable-next-line no-console
+	console.log( 'Hermes is: ' + isHermes() );
+
+	setupInitHooks();
+};
+
+const initializeGutenbergEditor = () =>
+	initializeEditor( 'gutenberg', 'post', 1 );
+
+const setupInitHooks = () => {
+	addAction( 'native.pre-render', 'core/react-native-editor', ( props ) => {
+		registerBlocks();
+
+		const capabilities = props.capabilities ?? {};
+		// Unregister non-supported blocks by capabilities
+		if (
+			getBlockType( 'core/block' ) !== undefined &&
+			capabilities.reusableBlock !== true
+		) {
+			unregisterBlockType( 'core/block' );
+		}
+	} );
+
+	// Map native props to Editor props
+	// TODO: normalize props in the bridge (So we don't have to map initialData to initialHtml)
+	addFilter(
+		'native.block_editor_props',
+		'core/react-native-editor',
+		( props ) => {
+			const { capabilities = {} } = props;
+			let {
+				initialData,
+				initialTitle,
+				postType,
+				featuredImageId,
+				colors,
+				gradients,
+				rawStyles,
+				rawFeatures,
+				galleryWithImageBlocks,
+				locale,
+			} = props;
+
+			if ( initialData === undefined && __DEV__ ) {
+				initialData = initialHtml;
+			}
+			if ( initialTitle === undefined ) {
+				initialTitle = 'Welcome to Gutenberg!';
+			}
+			if ( postType === undefined ) {
+				postType = 'post';
+			}
+
+			colors = validateThemeColors( colors );
+
+			gradients = validateThemeGradients( gradients );
+
+			return {
+				initialHtml: initialData,
+				initialHtmlModeEnabled: props.initialHtmlModeEnabled,
+				initialTitle,
+				postType,
+				featuredImageId,
+				capabilities,
+				colors,
+				gradients,
+				rawStyles,
+				rawFeatures,
+				galleryWithImageBlocks,
+				locale,
+			};
+		}
+	);
+};
+
+const registerBlocks = () => {
+	if ( blocksRegistered ) {
+		return;
+	}
+
+	registerCoreBlocks();
+
+	blocksRegistered = true;
+};
+
+export default () => {
+	if ( editorComponent ) {
+		return editorComponent;
+	}
+
+	reactNativeSetup();
+	gutenbergSetup();
+	editorComponent = initializeGutenbergEditor();
```

Comment:
```suggestion
	editorComponent = initializeEditor( 'gutenberg', 'post', 1 );
```
Is there a benefit to abstracting `initializeGutenbergEditor`? It would appear to only be utilized once and is fairly straightforward. WDYT about inlining the logic?

Reply:
Not really, I mainly abstracted it in case we would require further functionality there but I agree that since it's only one line we could inline it 👍 .

Reply:
I applied the suggestion in [this commit](https://github.com/WordPress/gutenberg/pull/37073/commits/e1f3cdc9cea1c7cc53ec6097c6788f54ff6c4b63).